janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,6 +506,117 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
+        } else analyzerString = null;
+
+        List<String> terms;
+        if (analyzerString != null) {
+            try {
+                final Class classDefinition = Class.forName(analyzerString);
+                final Analyzer analyzer = (Analyzer) classDefinition.newInstance();
+                terms = customTokenize(analyzer, key, value);
+            } catch (Exception e) {
+                throw new RuntimeException("Analyzer cannot be instanciated for class " + analyzerString, e);
+            }
+
+        } else {
+            if (mapping == null) mapping = Mapping.DEFAULT;
+            switch (mapping) {
+                case STRING:
+                    terms = Collections.singletonList(value);
+                    break;
+                case DEFAULT:
+                case TEXT:
+                    terms = Text.tokenize(value);
+                    break;
+                default:
+                    throw new RuntimeException("unsupported case " + mapping);
+            }
+
+        }
+        if (terms.isEmpty()) {
+            // This might happen with very short terms
+            if (janusgraphPredicate == Text.CONTAINS_PREFIX) {
+                final Term term;
+                if (mapping == Mapping.STRING) {
+                    term = new Term(key, value);
+                } else  {
+                    term = new Term(key, value.toLowerCase());
+                }
+                params.addQuery(new PrefixQuery(term), BooleanClause.Occur.MUST);
+            }
+        } else if (terms.size() == 1) {
+            if (janusgraphPredicate == Cmp.EQUAL || janusgraphPredicate == Text.CONTAINS) {
+                params.addQuery(new TermQuery(new Term(key, value)));
+            } else if (janusgraphPredicate == Cmp.NOT_EQUAL) {
+                BooleanQuery.Builder q = new BooleanQuery.Builder();
+                q.add(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
+                q.add(new TermQuery(new Term(key, value)), BooleanClause.Occur.MUST_NOT);
+                params.addQuery(q.build(), BooleanClause.Occur.MUST);
+            } else if (janusgraphPredicate == Text.CONTAINS_PREFIX) {
+                Term term;
+                if (mapping == Mapping.STRING) {
+                    term = new Term(key, terms.get(0));
+                } else  {
+                    term = new Term(key, terms.get(0).toLowerCase());
+                }
+                params.addQuery(new PrefixQuery(term), BooleanClause.Occur.MUST);
+            }
+        } else {
+            // at the moment, this is only walked for EQUAL and Text.CONTAINS (String and Text mappings)
+            BooleanQuery.Builder q = new BooleanQuery.Builder();
+            for (final String term : terms) {
+                q.add(new TermQuery(new Term(key, term)), BooleanClause.Occur.MUST);
+            }
+            params.addQuery(q.build());
+        }
+    }
+
+
+    public static String getHomeDir(String subDirectory) {
```

Comment:
I don't see any usages of this. Can it be removed?

Reply:
I don't know about that (I did not write this). 
I'll try and launch the lucene tests to see if it makes a difference

Reply:
The BerkeleyLuceneTest and LuceneIndexTest pass
So, it might be safe to remove this
I'll do it, next time

Reply:
I removed this